### PR TITLE
Screen Space Shadows Improvements

### PR DESCRIPTION
**Pull request by FrancescoC-unity**
### Purpose of this PR
This PR optimize screen space shadows (some basic ALU optimizations, early out conditions + tracing half res depth buffer until a hit is found). This optimization was measured as around a 30% win on Xbox One X. 
The quality of the effect has been increased as well, mainly by changing the noise function used to a time-animated Interleaved Gradient Noise. When TAA is off, the IGN is not animated to avoid flickering, however this results in lower quality.

Before and after with 8 samples per pixel
beforeafter

---
### Release Notes
Improving performance and quality of Screen Space Shadows.

---
### Testing status
**Katana Tests**: https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=ssshadow-improvements&automation-tools_branch=add-platform-filter&unity_branch=trunk

---
### Overall Product Risks
**Technical Risk**: Low

**Halo Effect**:  Low

---
### Comments to reviewers
I had to move all the changes to a new branch because of cryptic errors caused by a bad merge, I hope there is nothing fishy now.  Also, being my first PR I expect something wrong :-)

### CODE REVIEW

**Comment by FrancescoC-unity** [comment]
Katana finished all green

**Comment by FrancescoC-unity** [comment]
Merging broke contact shadows with my change, I am investigating now

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/ScreenSpaceShadow.compute [hunk 1]
```diff
@@ -37,88 +36,119 @@ CBUFFER_END
...
-    uint3 hashInput = uint3(abs(GetAbsolutePositionWS(positionWS)) * 1000);
+    UV.y *= (2.0f / 3.0f);
+
+    if (HalfRes)
+    {
+        UV.x = UV.x * 0.5f;
+        UV.y = UV.y * 0.5f + (2.0f / 3.0f);
+    }
+
+    return SAMPLE_TEXTURE2D_LOD(_CameraDepthTexture, sampler_CameraDepthTexture, _ScreenToTargetScale.xy * UV, 0).x;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/ScreenSpaceShadow.compute]: why using SAMPLE_TEXTURE2D_LOD? Should use just a lod, and no need for UV.y *= (2.0f / 3.0f);

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/ScreenSpaceShadow.compute.BASE.compute.meta [hunk 2]
```diff
@@ -0,0 +1,8 @@
+fileFormatVersion: 2
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/ScreenSpaceShadow.compute.BASE.compute.meta]: Remove this files

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/ScreenSpaceShadow.compute.LOCAL.compute.meta [hunk 3]
```diff
@@ -0,0 +1,8 @@
+fileFormatVersion: 2
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/ScreenSpaceShadow.compute.LOCAL.compute.meta]: Remove this files

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/ScreenSpaceShadow.compute.REMOTE.compute.meta [hunk 4]
```diff
@@ -0,0 +1,8 @@
+fileFormatVersion: 2
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/ScreenSpaceShadow.compute.REMOTE.compute.meta]: Remove this files

**Comment by FrancescoC-unity** [comment]
Note that contact shadows will be broken until [this PR](https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/2071) gets merged

**Pull request merged**